### PR TITLE
ppx: drop special encoding of enum like variants

### LINKED ISSUE #25: encoding of enumeration like variants should be an array
https://github.com/melange-community/melange-json?tab=readme-ov-file#enumeration-like-variants

This behavior is not consistent with ppx_deriving_yojson/ppx_yojson_conv. But more importantly to me it means that adding a new constructor to a variant might make it impossible to read a json generated previously, which is pretty unexpected.

```ocaml
# type t = Aaa [@@deriving json];;
type t = Aaa
val of_json : Ppx_deriving_json_runtime.t -> t = <fun>
val to_json : t -> Ppx_deriving_json_runtime.t = <fun>
# Aaa |> to_json |> Ppx_deriving_json_runtime.to_string |> print_endline;;
"Aaa"
(* extending the type, no longer an enum *)
# type t = Aaa | Bbb of int [@@deriving json];;
type t = Aaa | Bbb of int
val of_json : Ppx_deriving_json_runtime.t -> t = <fun>
val to_json : t -> Ppx_deriving_json_runtime.t = <fun>
# Aaa |> to_json |> Ppx_deriving_json_runtime.to_string |> print_endline;;
["Aaa"]
# `List [ `String "Aaa" ] |> of_json;;
- : t = Aaa
# `String "Aaa" |> of_json;;
Exception: Ppx_deriving_json_runtime.Of_json_error "invalid JSON".
```

### PR DESCRIPTION
**Pull request by Khady**
Fixes #25

I need to update the tests but have a problem with my setup

### CODE REVIEW

File: ppx/test/example.ml [hunk 1]
```diff
@@ -40,10 +40,6 @@ module Cases = struct
     C ({|["B", 42]|}, poly_of_json, poly_to_json, (`B 42 : poly));
     C ({|["Fix",["Fix",["Fix",["A"]]]]|}, recur_of_json, recur_to_json, (Fix (Fix (Fix A))));
     C ({|["Fix",["Fix",["Fix",["A"]]]]|}, polyrecur_of_json, polyrecur_to_json, (`Fix (`Fix (`Fix `A))));
-    C ({|"A"|}, evar_of_json, evar_to_json, (A : evar));
-    C ({|"b_aliased"|}, evar_of_json, evar_to_json, (B : evar)); (* variant B repr as "b_aliased" in JSON *)
-    C ({|"b"|}, epoly_of_json, epoly_to_json, (`b : epoly));
-    C ({|"A_aliased"|}, epoly_of_json, epoly_to_json, (`a : epoly)); (* polyvariant `a aliased to "A_aliased"*)
```
Comment (Khady) [inline ppx/test/example.ml:46]: no longer supported

File: ppx/test/ppx_deriving_json_js.e2e.t [hunk 2]
```diff
@@ -60,14 +60,6 @@
...
   JSON    DATA: ["Fix",["Fix",["Fix",["A"]]]]
   JSON REPRINT: ["Fix",["Fix",["Fix",["A"]]]]
-  JSON    DATA: "A"
-  JSON REPRINT: "A"
-  JSON    DATA: "b_aliased"
-  JSON REPRINT: "b_aliased"
-  JSON    DATA: "b"
-  JSON REPRINT: "b"
-  JSON    DATA: "A_aliased"
-  JSON REPRINT: "A_aliased"
```
Comment (Khady) [inline ppx/test/ppx_deriving_json_js.e2e.t:70]: no longer supported

**Review by andreypopp — APPROVED** [review]
thank you!

**Comment by andreypopp** [comment]
need to update README as well

File: ppx/test/example.ml (hunk 1 above)
Comment (anmonteiro) [reply ppx/test/example.ml:46]: can't we keep the test cases and print the fixed representation, or were they redundant?

**Review by jchavarri — APPROVED** [review]
Thanks! I think this should have a changelog entry as well.

**Comment by andreypopp** [comment]
> Thanks! I think this should have a changelog entry as well.

done, and rebased

**Pull request merged**